wordpress-mobile/WordPress-Android
# Issue/10642 material plans and domains

Fixes #10642 

This PR moves Plans and Domain Registration screens to Material Theme and adds Dark Theme support.

Plans is a really small part of the app, so I bundled it with Domain Registration flow.

While refactoring is not in the scope of this PR, since I touched those files, I cleaned up/optimized some of the redundant code as suggested by AS, so do not be alarmed by the size of the PR. I suggest turning on the "Hide whitespace changes" option during the review.

There is a huge amount of PRs in this feature that covers all the screens within the app, so I'm requesting reviews from a wide variety of people.

![comparison_plans](https://user-images.githubusercontent.com/728822/71947730-a27f9080-3182-11ea-8876-f36e3bd3830c.png)
![comparison_domains](https://user-images.githubusercontent.com/728822/71947738-a8757180-3182-11ea-85a0-92ae9e78b9ed.png)


To test:
- Navigate to Plans screen and confirm it's visual and functional integrity in both Light and Dark themes. 
- Using a site with unclaimed domain credit, Navigate to Domain Registration screen and confirm it's visual and functional integrity in both Light and Dark themes. There are no functional changes to domain registration flow, so you might choose not to register an actual domain name, instead, you can add following line `mainViewModel.completeDomainRegistration(DomainRegistrationCompletedEvent("super-test-domain.com", "super-test-email@wordpress.org"))` 
to the end of the method like `onViewCreated`  in `DomainRegistrationDetailsFragment` to skip to the final screen.
The theme can be changed from App Settings.

PR submission checklist:
